framework/src/utils/RankThreeTensor.C

```diff
@@ -23,6 +23,11 @@
 #include <iomanip>
 #include <ostream>
 
+#define r3i(i, j, k) ((i * LIBMESH_DIM + j) * LIBMESH_DIM + k)
+#define N LIBMESH_DIM
+#define N2 (LIBMESH_DIM * LIBMESH_DIM)
+#define N3 (LIBMESH_DIM * LIBMESH_DIM * LIBMESH_DIM)
```

Comment:
They are only in the C file.  It makes sure we don't waste time allocating variables, loading values from other memory locations, or doing extra math at runtime.